### PR TITLE
Upgrade docker containers

### PR DESCRIPTION
**Pull request by chapman39**
* Create images from [radiuss-docker](https://github.com/LLNL/radiuss-docker/pkgs/container/radiuss) instead of axom-docker
* Upgrade compilers to clang15 and gcc13
* Upgrade OS to ubuntu22
* Update `Doxyfile.in` to 1.9.1

### CODE REVIEW

File: src/docs/sphinx/dev_guide/new_docker_image.rst [hunk 1]
```diff
@@ -16,14 +16,9 @@ Create New Docker File
...
-   created before proceeding. Here is the `Axom Docker repository <https://github.com/LLNL/axom-docker>`_.
-
-   .. note:: `A new docker repository <https://github.com/rse-ops/docker-images>`_ has been created and
-      is being actively maintained for RADIUSS unlike the Axom Docker repo. We should migrate to it but it may
-      be missing system packages.
-
+#. Ensure that an image exists on Dockerhub for the desired compiler.
+   If no corresponding compiler image exists, it should be 
+   created before proceeding. Here is the `RADIUSS Docker repository images <https://github.com/LLNL/radiuss-docker/pkgs/container/radiuss>`_.
 #. Go to the ``scripts/docker`` directory and run ``build_new_dockerfile.sh``, passing the compiler
```
Comment (chapman39) [inline src/docs/sphinx/dev_guide/new_docker_image.rst:22]: I need to update this still

**Comment by chapman39** [comment]
@samuelpmish @white238 @jamiebramwell I'm not sure if anyone's familiar with this error, but I'm having trouble how to approach it

```
[ 17%] Linking CXX static library ../../../../lib/libserac_contact.a
cd /home/serac/serac/_serac_build_and_test_2024_02_26_16_51_55/build-gcc@13.1.0/src/serac/physics/contact && /usr/local/bin/cmake -P CMakeFiles/serac_contact.dir/cmake_clean_target.cmake
cd /home/serac/serac/_serac_build_and_test_2024_02_26_16_51_55/build-gcc@13.1.0/src/serac/physics/contact && /usr/local/bin/cmake -E cmake_link_script CMakeFiles/serac_contact.dir/link.txt --verbose=1
/usr/bin/ar qc ../../../../lib/libserac_contact.a CMakeFiles/serac_contact.dir/contact_data.cpp.o CMakeFiles/serac_contact.dir/contact_interaction.cpp.o
/usr/bin/ranlib ../../../../lib/libserac_contact.a
make[2]: Leaving directory '/home/serac/serac/_serac_build_and_test_2024_02_26_16_51_55/build-gcc@13.1.0'
[ 17%] Built target serac_contact
In file included from /usr/include/c++/13/bits/stl_uninitialized.h:63,
                 from /usr/include/c++/13/memory:69,
                 from /home/serac/serac/src/serac/infrastructure/../../serac/mesh/mesh_utils_base.hpp:16,
                 from /home/serac/serac/src/serac/infrastructure/../../serac/mesh/mesh_utils.hpp:16,
                 from /home/serac/serac/src/serac/mesh/mesh_utils.cpp:7:
In static member function ‘static _Up* std::__copy_move<_IsMove, true, std::random_access_iterator_tag>::__copy_m(_Tp*, _Tp*, _Up*) [with _Tp = const double; _Up = double; bool _IsMove = false]’,
    inlined from ‘_OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = const double*; _OI = double*]’ at /usr/include/c++/13/bits/stl_algobase.h:506:30,
    inlined from ‘_OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = const double*; _OI = double*]’ at /usr/include/c++/13/bits/stl_algobase.h:533:42,
    inlined from ‘_OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = const double*; _OI = double*]’ at /usr/include/c++/13/bits/stl_algobase.h:540:31,
    inlined from ‘_OI std::copy(_II, _II, _OI) [with _II = const double*; _OI = double*]’ at /usr/include/c++/13/bits/stl_algobase.h:633:7,
    inlined from ‘static _ForwardIterator std::__uninitialized_copy<true>::__uninit_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = const double*; _ForwardIterator = double*]’ at /usr/include/c++/13/bits/stl_uninitialized.h:147:27,
    inlined from ‘_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = const double*; _ForwardIterator = double*]’ at /usr/include/c++/13/bits/stl_uninitialized.h:185:15,
    inlined from ‘_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = const double*; _ForwardIterator = double*; _Tp = double]’ at /usr/include/c++/13/bits/stl_uninitialized.h:373:37,
    inlined from ‘void std::vector<_Tp, _Alloc>::_M_assign_aux(_ForwardIterator, _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = const double*; _Tp = double; _Alloc = std::allocator<double>]’ at /usr/include/c++/13/bits/vector.tcc:345:35,
    inlined from ‘std::vector<_Tp, _Alloc>& std::vector<_Tp, _Alloc>::operator=(std::initializer_list<_Tp>) [with _Tp = double; _Alloc = std::allocator<double>]’ at /usr/include/c++/13/bits/stl_vector.h:787:21,
    inlined from ‘serac::mesh::InputOptions FromInlet<serac::mesh::InputOptions>::operator()(const axom::inlet::Container&)’ at /home/serac/serac/src/serac/mesh/mesh_utils.cpp:494:58:
/usr/include/c++/13/bits/stl_algobase.h:437:30: error: ‘void* __builtin_memcpy(void*, const void*, long unsigned int)’ forming offset 24 is out of the bounds [0, 24] [-Werror=array-bounds=]
  437 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

It's failing only gcc13-release. https://dev.azure.com/llnl-serac/serac/_build/results?buildId=10908&view=logs&j=8418a147-cefe-5fbf-eb11-e069d1299bbe&t=02db9b20-7613-5b66-5602-b48459f634fd&l=463

Edit: I figured it out.

File: scripts/docker/dockerfile.in [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
 
+# Remove flags from mpich
+RUN sudo sed -i "s/ -flto=auto//g" /usr/bin/mpicxx.mpich
+RUN sudo sed -i "s/ -flto=auto//g" /usr/bin/mpicc.mpich
+RUN sudo sed -i "s/ -flto=auto//g" /usr/bin/mpifort.mpich
+RUN sudo sed -i "s/ -ffat-lto-objects//g" /usr/bin/mpicxx.mpich
+RUN sudo sed -i "s/ -ffat-lto-objects//g" /usr/bin/mpicc.mpich
+RUN sudo sed -i "s/ -ffat-lto-objects//g" /usr/bin/mpifort.mpich
+RUN sudo sed -i "s/ -fallow-invalid-boz//g" /usr/bin/mpifort.mpich
+RUN sudo sed -i "s/ -fallow-argument-mismatch//g" /usr/bin/mpifort.mpich
```
Comment (chapman39) [inline scripts/docker/dockerfile.in:19]: May be worth bringing this up to radiuss-docker, if they end up building clang with mpich.

File: scripts/docker/dockerfile_clang-15 [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
 USER serac
 
 RUN git clone --recursive --branch $branch --single-branch --depth 1 https://github.com/LLNL/serac.git serac_repo
 
 # Build/install TPLs via spack and then remove the temporary build directory on success
-RUN cd serac_repo && python3 ./scripts/uberenv/uberenv.py --spack-env-file=./scripts/spack/configs/docker/ubuntu20/spack.yaml \
+RUN cd serac_repo && python3 ./scripts/uberenv/uberenv.py --spack-env-file=./scripts/spack/configs/docker/ubuntu22/spack.yaml \
                                                           --project-json=.uberenv_config.json \
-                                                          --spec=%clang@10.0.0~strumpack --prefix=/home/serac/serac_tpls -k \
+                                                          --spec=%clang@15.0.7 --prefix=/home/serac/serac_tpls -k \
```
Comment (chapman39) [inline scripts/docker/dockerfile_clang-15:29]: Strumpack works on docker containers now

File: scripts/docker/dockerfile.in (hunk 2 above)
Comment (white238) [reply scripts/docker/dockerfile.in:19]: @davidbeckingsale would you like this to go into the base radiuss image? The mpich wrappers add link time optimization flags which then fails in the non-gcc compilers due to it being very specific.

File: scripts/llnl/common_build_functions.py [hunk 4]
```diff
@@ -271,7 +271,7 @@ def build_and_test_host_config(test_root, host_config, report_to_stdout=False, e
     bld_output_file =  pjoin(build_dir,"output.log.make.txt")
     print("[starting build]")
     print("[log file: %s]" % bld_output_file)
-    res = sexe("cd %s && make -j 8 VERBOSE=1 " % build_dir,
+    res = sexe("cd %s && make -j4 VERBOSE=1 " % build_dir,
```
Comment (white238) [inline scripts/llnl/common_build_functions.py]: did this fail?

Comment (chapman39) [reply scripts/llnl/common_build_functions.py]: I thought maybe it was related to that gcc array-bounds error I was getting, but I can likely revert this back now.

**Review by white238 — APPROVED** [review]
Thanks @chapman39 ! Did this pass the style check w/o any changes? That would be amazing since it was so version dependent before.

**Comment by white238** [comment]
@chapman39 can you fix the following warning:

```
CMake Warning at cmake/blt/cmake/SetupCodeChecks.cmake:695 (message):
  blt_add_clangformat_target: clang-format '10'' is required, found '15.0.7'.
  Disabling 'style' build target.
Call Stack (most recent call first):
  cmake/blt/cmake/SetupCodeChecks.cmake:206 (blt_add_clangformat_target)
  cmake/SeracMacros.cmake:50 (blt_add_code_checks)
  CMakeLists.txt:167 (serac_add_code_checks)
```

https://github.com/LLNL/serac/blob/7672d330192aec5f4eb46ef48e0f5ec713b203f7/CMakeLists.txt#L104

**Review by jamiebramwell — APPROVED** [review]
Thanks @chapman39 !

**Pull request merged**